rpcs3/Emu/System.cpp

```diff
@@ -2953,24 +3016,37 @@ void Emulator::Kill(bool allow_autoexit, bool savestate, savestate_stage* save_s
 
 		if (savestate)
 		{
-			const std::string path = get_savestate_file(m_title_id, m_path, 0, 0);
-
-			if (!fs::create_path(fs::get_parent_dir(path)))
-			{
-				sys_log.error("Failed to create savestate directory! (path='%s', %s)", fs::get_parent_dir(path), fs::g_tls_error);
-			}
-
-			fs::pending_file file(path);
-
 			// Identifer -> version
 			std::vector<std::pair<u16, u16>> used_serial = read_used_savestate_versions();
 
 			auto& ar = *to_ar;
+
```

Comment:
Can't you put this entire scope into the utils or some member func as well?
Idk, like ar.save_to_file() or whatever.

Reply:
Fixed.